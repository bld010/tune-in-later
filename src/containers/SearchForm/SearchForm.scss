.SearchForm {

  & form {
    display: flex;
    flex-direction: row;
    margin: 10px 0px 0px 15px;
  }
      & input{
        width: 200px;
        margin: 0;
      }

      & input[type="image"]{
        width: 30px;
        margin: 0px 0px 0px 15px;
      }
      
      & button {
        margin: 0;
      }
  
  & img {
    width: 30px;
    height: 30px;
    margin: 0
  }
  
}