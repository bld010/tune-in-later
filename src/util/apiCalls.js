export const landingFetch = () => {
  const romanceFetch = fetch('https://itunes.apple.com/search?media=audiobook&term=romance');
  const fantasyFetch = fetch('https://itunes.apple.com/search?media=audiobook&term=fantasy');
  const biographyFetch = fetch('https://itunes.apple.com/search?media=audiobook&term=biography');
  const historyFetch = fetch('https://itunes.apple.com/search?media=audiobook&term=history');
  const horrorFetch = fetch('https://itunes.apple.com/search?media=audiobook&term=horror');

  return Promise.all([romanceFetch, fantasyFetch, biographyFetch, historyFetch, horrorFetch])
    .then(responses => Promise.all(responses.map(response => response.json())))
    .then(data => {
      return data.map((datum, i) => {
        return datum.results.map(obj => {
          switch (i) {
            case 0:
              return { ...obj, filterType: 'romance' };
            case 1:
              return { ...obj, filterType: 'fantasy' };
            case 2:
              return { ...obj, filterType: 'biography' };
            case 3:
              return { ...obj, filterType: 'history' };
            case 4:
              return { ...obj, filterType: 'horror' };
            default:
              return obj;
          }
          })
        })
      })
    .catch(err => console.log(err, 'error in apiCalls landingFetch'))

}

export const authorFetch = (authorFirstName, authorLastName) => {
  return fetch(`https://itunes.apple.com/search?media=audiobook&term=${authorFirstName+authorLastName}`)
    .then(response => response.json())
    .catch(err => console.log(err, 'error in apiCalls authorFetch'))

}

export const loginUser = (email, password) => {
  const user = {
    "email": email,
    "password": password
  };

  const options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  }

  return fetch('http://localhost:3001/api/v1/login/', options)    
}

export const addFavoritesToApi = (book, userId) => {
  const favoriteBook = {
    "book_id": book.id,
    "author_name": book.artist,
    "book_name": book.title,
     "artwork_url": book.image,
    "release_date": book.date,
    "description": book.description
  }
  const options = {
    method: "POST",
    body: JSON.stringify(favoriteBook),
    headers: {
      "Content-Type": "application/json"
    }
  }
  return fetch(`http://localhost:3001/api/v1/users/${userId}/bookfavorites`, options)
}