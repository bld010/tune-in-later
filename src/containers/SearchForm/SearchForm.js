import React, { Component } from 'react';
import { connect } from 'react-redux';
import './SearchForm.scss';
import magnifierTool from '../../images/magnifier-tool.svg';
import { fetchSearch } from '../../util/apiCalls';
import { displaySearchResults } from '../../actions/';
import PropTypes from 'prop-types';

export class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchTerm: '',
      results: []
    }
  }


  handleSearchInput = (e) => {
    this.setState({ searchTerm: e.target.value})
  }


  clearSearch = () => {
    this.setState({ searchTerm: '' })
  }

  handleSearch = (e) => {
    e.preventDefault();
    this.props.displaySearchResults([]);
    fetchSearch(this.state.searchTerm)
      .then(resp => {
        const results = this.cleanSearchResults(resp.results)
        this.props.displaySearchResults(results)
      }
      )
    this.clearSearch();
  }

  cleanSearchResults = (rawSearchResults) => {
    const allBooks = rawSearchResults.map(datum => datum).flat();
    let regex = new RegExp('(&nbsp;|<([^>]+)>)', 'g');
    const formattedBooks = allBooks.map(book => ({
      id: book.collectionId,
      date: book.releaseDate,
      artist: book.artistName,
      image: book.artworkUrl100,
      price: book.collectionPrice,
      title: book.collectionName,
      genre: book.primaryGenreName,
      description: book.description.replace(regex, ''),
      filterType: book.filterType
    }));
    return formattedBooks
  }

  render() {
 
    return(
      <div className="SearchForm">
        <form>
          <input
            type="text"
            value={this.state.searchTerm}
            placeholder="Search"
            onChange={this.handleSearchInput}
            ></input>
          <input 
            type="image" 
            src={magnifierTool}
            alt="search"
            onClick={this.handleSearch}
            ></input>
        </form>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  displaySearchResults: (searchResults) => dispatch(displaySearchResults(searchResults))
})

export default connect(null, mapDispatchToProps)(SearchForm);

SearchForm.propTypes = {
  displaySearchResults: PropTypes.func.isRequired
}