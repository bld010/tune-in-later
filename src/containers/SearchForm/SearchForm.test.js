import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm } from './SearchForm';
import { fetchSearch } from '../../util/apiCalls';

jest.mock('../../util/apiCalls');

describe('SearchForm', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
    <SearchForm 
      displaySearchResults={jest.fn()}
    />)
  })
  it('should match the snapshot with correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('handleSearchInput', () => {

  })

  describe('clearSearch', () => {

  })

  describe('handleSearch', () => {

  })

  describe('cleanSearchResults', () => {

  })

  describe('matchDispatchToProps', () => {
    
  })

})