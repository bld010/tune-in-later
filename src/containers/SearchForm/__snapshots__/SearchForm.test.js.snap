// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with correct data passed in 1`] = `
<div
  className="SearchForm"
>
  <form>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <input
      alt="search"
      onClick={[Function]}
      src="magnifier-tool.svg"
      type="image"
    />
  </form>
</div>
`;
